server/pulp/plugins/conduits/repo_sync.py

```diff
@@ -189,3 +190,21 @@ def build_cancel_report(self, summary, details):
                        self._removed_count, summary, details)
         r.canceled_flag = True
         return r
+
+    def last_sync(self):
+        """
+        Returns the timestamp of the last time this repo was successfully synced. If the repo
+        was never synced, this call returns None.
+
+        :return: timestamp instance describing the last sync
+        :rtype:  datetime.datetime or None
+
+        :raises ImporterConduitException: if any errors occur
+        """
+        try:
+            importer = model.Importer.objects.only('last_sync').get_or_404(
+                id=self.importer_object_id)
+            return importer.last_sync
+        except Exception as e:
+            _logger.exception('Error getting last sync time for repo [%s]' % self.repo_id)
+            raise ImporterConduitException(e), None, sys.exc_info()[2]
```

Comment:
I know this is the pattern found elsewhere in this file, but one of the painful things about Pulp is the many places that it obfuscates Exceptions like this. I think it would be better to just let the Exception raise (i.e., just drop the try/except).